nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -313,6 +327,12 @@ private static boolean isDirectTypeUseAnnotation(Attribute.TypeCompound t) {
           locationHasInnerTypes = true;
           break;
         case ARRAY:
+          // Currently we are ignoring @Nullable annotations on type in JSpecify mode.
+          // Eventually, this should return true, and array access should be handled
+          // elsewhere.
```

Comment:
Why do we say this should eventually return `true`?  I added some docs to the enclosing method, and I think what it is meant to do is determine whether an annotation applies to a top-level type.  Given that, shouldn't this method always return `false` in this spot in JSpecify mode?

Reply:
I think this and all subsequent comments are attributed to the fallacy that it works the other way around. All of these are actually working as we want them to.